### PR TITLE
Configuration overhaul (support environment variables)

### PR DESCRIPTION
**Pull request by callahad**
Supersedes #48.

I believe all of the commits are sane and sufficiently granular to satisfy the reviews on the former.

### CODE REVIEW

**Comment by callahad** [comment]
Ugh, rebasing botched a commit. Re-checking everything.

**Comment by callahad** [comment]
Every commit builds. Ready for review.

File: src/config.rs [hunk 1]
```diff
@@ -249,6 +250,39 @@ impl ConfigBuilder {
...
+        if let Some(val) = env_config.broker_keyfiles { self.keyfiles = val; }
+
+        if let Some(val) = env_config.broker_redis_url { self.redis_url = Some(val); }
+        if let Some(val) = env_config.broker_session_ttl { self.redis_session_ttl = val; }
+        if let Some(val) = env_config.broker_cache_ttl { self.redis_cache_ttl = val; }
+        if let Some(val) = env_config.broker_cache_max_doc_size { self.redis_cache_max_doc_size = val; }
+
+        if let Some(val) = env_config.broker_from_name { self.from_name = val; }
+        if let Some(val) = env_config.broker_from_address { self.from_address = Some(val); }
+        if let Some(val) = env_config.broker_smtp_server { self.smtp_server = Some(val); }
```
Comment (stephank) [inline src/config.rs]: This is missing username / password?

File: config.toml.dist [hunk 2]
```diff
@@ -1,28 +1,59 @@
+[server]
+# IP address to bind to - Default: "127.0.0.1"
 listen_ip = "127.0.0.1"
+
```
Comment (stephank) [inline config.toml.dist]: I think this'd be much more readable if we just have a newline between sections, not every field.

File: src/config.rs [hunk 3]
```diff
@@ -132,45 +170,62 @@ impl Config {
             ));
         }
```
Comment (stephank) [inline src/config.rs]: This should move to `done` as well.

File: config.toml.dist (hunk 2 above)
Comment (callahad) [inline config.toml.dist]: I could've sworn I did that :/ redoing

**Comment by callahad** [comment]
Wrapped up changes on my flight, will push as soon as I get home (~45 min)

**Comment by callahad** [comment]
Ready for re-review

**Comment by stephank** [comment]
Splendid! Solid engineering, sir! ✔️

**Pull request merged**